tests/base.py

```diff
@@ -0,0 +1,2602 @@
+           build_args=DEFAULT_BUILD_ARGS),
+    Binary(name='acra-tokens', from_version=DEFAULT_VERSION,
+           build_args=DEFAULT_BUILD_ARGS),
+]
+
+BUILD_TAGS = os.environ.get("TEST_BUILD_TAGS", '')
+
+
+def build_binaries():
+    """Build Acra CE binaries for testing."""
+    builds = [
+        (binary.from_version, ['go', 'build', '-o={}'.format(os.path.join(BINARY_OUTPUT_FOLDER, binary.name)),
+                               '-tags={}'.format(BUILD_TAGS)] +
+         binary.build_args +
+         ['github.com/cossacklabs/acra/cmd/{}'.format(binary.name)])
+        for binary in BINARIES
+    ]
+    go_version = get_go_version()
+    GREATER, EQUAL, LESS = (1, 0, -1)
+    for version, build in builds:
+        if semver.VersionInfo.parse(go_version).compare(version) == LESS:
+            continue
+        # try to build 3 times with timeout
+        build_count = 3
+        for i in range(build_count):
+            try:
+                subprocess.check_call(build, cwd=os.getcwd(), timeout=PROCESS_CALL_TIMEOUT)
+                break
+            except (AssertionError, subprocess.TimeoutExpired):
+                if i == (build_count - 1):
+                    raise
+                continue
+
+
+def clean_binaries():
+    for i in BINARIES:
+        try:
+            os.remove(os.path.join(BINARY_OUTPUT_FOLDER, i.name))
+        except:
+            pass
+
+
+def clean_misc():
+    pass
+
+
+PROCESS_CALL_TIMEOUT = 120
+
+
+def get_go_version():
+    output = subprocess.check_output(['go', 'version'])
+    # example: go1.7.2 or go1.7
+    version = re.search(r'go([\d.]+)', output.decode('utf-8')).group(1)
+    # convert to 3 part semver format
+    if version.count('.') < 2:
+        version = '{}.0'.format(version)
+    return version
+
+
+def drop_tables():
+    engine_raw = sa.create_engine(
+        '{}://{}:{}/{}'.format(DB_DRIVER, DB_HOST, DB_PORT, DB_NAME),
+        connect_args=connect_args)
+    metadata.drop_all(engine_raw)
+    engine_raw.dispose()
+
+
+# Set this to False to not rebuild binaries on setup.
+CLEAN_BINARIES = utils.get_bool_env('TEST_CLEAN_BINARIES', default=True)
+# Set this to False to not build binaries in principle.
+BUILD_BINARIES = True
+
+hasSetUp = False
+
+
+def collectTests(source: unittest.TestSuite, tests=[]):
+    if not isinstance(source, unittest.TestSuite):
+        raise TypeError('invalid source of tests')
+    for tcase in source._tests:
+        if not tcase:
+            continue
+        if isinstance(tcase, unittest.TestSuite):
+            collectTests(tcase, tests)
+        if isinstance(tcase, unittest.TestCase):
+            tests.append(tcase)
+
+
+def load_tests(loader, standard_tests, pattern):
+    tests = []
+    collectTests(standard_tests, tests)
+    for tcase in tests:
+        if not tcase:
+            continue
+        tcase.__class__.__module__ = 'base'
+    return unittest.TestSuite(tests)
+
+
+def setUpModule():
+    baseSetUpModule()
```

Comment:
do we need separate function with different name or we can leave only `setUpModule`?

and what about to move these functions to the bottom of the file to simplify finding callback? and lets add comment with reference to the documentation that we patch module names due to unittest runs test cases one by one and calls setUpModule once only if test cases ordered by module name or from one module, otherwise if tests unordered with same modules, setUpModule will be called same times. So we specify one name for all of them because need only one call and only one such method we use. Otherwise, in the future, we should order by `.__class__.__module__` testcases to allow call setUpModule per modules.